### PR TITLE
Apex for RUB-214

### PR DESCRIPTION
**Pull request by ishwari-gaikwad2**

### CODE REVIEW

File: force-app/main/default/classes/ApplicationTriggerHandler.cls [hunk 1]
```diff
@@ -51,4 +51,71 @@ public class ApplicationTriggerHandler {
...
+      }
+    }
+  }
+
+  /**
+   * @description : This method is used to get map of group from salesforce
+   * @author Ishwari Gaikwad(Thinqloud) | 05-18-2021
+   * @return Map<String, Group>
+   **/
+  private static Map<String, Group> getQueueMap() {
```
Comment (RemarioCRM) [inline force-app/main/default/classes/ApplicationTriggerHandler.cls:84]: use this getQueueId from Util class instead

File: force-app/main/default/classes/ApplicationTriggerHandler.cls [hunk 2]
```diff
@@ -51,4 +51,71 @@ public class ApplicationTriggerHandler {
       }
     }
   }
+
+  /**
+   * @description : This method is used when the application is accepted from the ADJ
+   * @author Ishwari Gaikwad(Thinqloud) | 05-18-2021
+   **/
+  public static void changeApplicationStatus() {
+    Map<String, Group> groupMap = getQueueMap();
```
Comment (RemarioCRM) [inline force-app/main/default/classes/ApplicationTriggerHandler.cls:60]: use getQueueId from Util

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/triggers/ApplicationTrigger.trigger [hunk 3]
```diff
@@ -8,10 +8,10 @@
  * Ver   Date         Author             Modification
  * 1.0   04-20-2021   Remario Richards   Initial Version
  **/
-trigger ApplicationTrigger on Application__c(after insert, before update) {
+trigger ApplicationTrigger on Application__c(before insert, before update) {
   ApplicationTriggerHandler.init(Trigger.new, Trigger.oldMap, Trigger.newMap);
   if (Trigger.isInsert) {
-    if (Trigger.isAfter) {
+    if (Trigger.isBefore) {
```
Comment (RemarioCRM) [inline force-app/main/default/triggers/ApplicationTrigger.trigger:14]: why did you change this @ishwari-gaikwad2

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/triggers/ApplicationTrigger.trigger (hunk 3 above)
Comment (ishwari-gaikwad2) [reply force-app/main/default/triggers/ApplicationTrigger.trigger:14]: We have validation on application creation that opportunity will not have more than 2 applications. but that was not working as that trigger was on after insert. so I have changed it.

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

**Comment by RemarioCRM** [comment]
merge and resolve all conflicts

File: force-app/main/default/classes/ApplicationTriggerHandler.cls [hunk 4]
```diff
@@ -44,11 +44,60 @@ public class ApplicationTriggerHandler {
       if (opportunityMap.containsKey(newAppRecord.Opportunity_ID__c)) {
         if (
           opportunityMap.get(newAppRecord.Opportunity_ID__c)
-            .Applications__r.size() > 1
+            .Applications__r.size() >= 1
```
[the remark this answers is not in the document]
Comment (ishwari-gaikwad2) [reply force-app/main/default/classes/ApplicationTriggerHandler.cls:47]: due to requirement of "opportunity will have only one applicant record "

File: force-app/main/default/triggers/ApplicationTrigger.trigger [hunk 5]
```diff
@@ -2,17 +2,23 @@
...
       ApplicationTriggerHandler.checkOpportunityHasOneApplication();
     }
   }
+
+  if (Trigger.isUpdate) {
```
Comment (RemarioCRM) [inline force-app/main/default/triggers/ApplicationTrigger.trigger:19]: ensure this trigger. has top level comments

File: force-app/main/default/classes/ApplicationTriggerHandler.cls [hunk 6]
```diff
@@ -2,8 +2,8 @@
  * @description       : To Validate that one opportunity have one application
  * @author            : Tushar Chopade(Thinqloud)
  * @group             :
- * @last modified on  : 04-21-2021
- * @last modified by  : Remario Richards
+ * @last modified on  : 05-19-2021
+ * @last modified by  : Ishwari Gaikwad(Thinqloud)
```
Comment (RemarioCRM) [inline force-app/main/default/classes/ApplicationTriggerHandler.cls:6]: Please. pull locally, review and merge then resolve conflicts. Repush afterwards

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

**Pull request closed without merging**